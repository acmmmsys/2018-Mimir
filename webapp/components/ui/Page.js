import React, { Component } from 'react';
import classnames from 'classnames';

import 'style/ui/Page.scss'

const defaultClassNames = [
    'page',
];

class Page extends Component {
   
    static defaultProps = {
        classNames: [],
    }

    render() {
        let classNames = classnames(
            defaultClassNames,
            this.props.classNames,
        );

        return (
            <div className={classNames}>
                {this.props.children}
            </div>
        );
    }
}

export default Page;