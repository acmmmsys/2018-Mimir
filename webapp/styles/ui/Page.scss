$scale: .78;

h4 {
    padding: 0;
    margin-top: 0;
}

.page-header {
    display: flex;
    flex-direction: row;
    width: 100%;
}

.page {
    background: white;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    padding: 25px;
    margin-bottom: 0.5cm;
    box-shadow: 0 0 0.5cm rgba(0,0,0,0.5);
    width: 21cm * $scale;
    height: 29.7cm * $scale;
    box-sizing: border-box;

    header {
        align-self: flex-start;
    }

    textarea {
        resize: none;
    }
}

.input-box-group {    
    label {
        display: inline-block;
        width: 230px!important;
        text-align: right;
        padding-right: 10px;
    }
}

.input-group {
    width: 100%;
    padding: 1px 0;

    label {
        display: inline-block;
        width: 150px;
        text-align: right;
        padding-right: 10px;
    }

    input[type="text"] {
        border: none;
        border-bottom: solid 2px white;

        &:focus {
            outline: none;
            border: none;
            border-bottom: solid 2px #c9c9c9;
        }
    }
}