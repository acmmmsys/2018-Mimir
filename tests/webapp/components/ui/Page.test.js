import React from 'react';
import Page from '../../../../webapp/components/ui/Page'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('Page does render', () => {
  const wrapper = shallow(
      <Page store={mockStore(initialState)}>This is a test!</Page>
  );
  expect(wrapper).toMatchSnapshot();
});
