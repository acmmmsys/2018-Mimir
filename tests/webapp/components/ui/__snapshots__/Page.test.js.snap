// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page does render 1`] = `
<div
  className="page "
>
  This is a test!
</div>
`;
